> A large majority of changes is quotes around env var usage, and those seem spurious, things work without quoting and without is the natural way to write such code by hand. Am I missing something? If not, can that rule be turned off?

The rule is https://www.shellcheck.net/wiki/SC2086. I don't feel strongly if you would rather suppress those warnings.

--- a/.github/actionlint.yml
+++ b/.github/actionlint.yml
@@ -2,3 +2,7 @@ paths:
   .github/workflows/gpu-ci.yml:
     ignore:
       - 'label "ghcr.io/cirruslabs/ubuntu-runner-amd64-gpu:24.04" is unknown'
+  .github/workflows/**/*.{yml,yaml}:
+    ignore:
+      # ignore double quoting code SC2086
+      - 'shellcheck reported issue in this script: SC2086:.+'

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -394,20 +394,23 @@ jobs:
         run: |
           set -exuo pipefail
           docker pull quay.io/pypa/manylinux_2_28_i686
-          docker run -v $(pwd):/scipy --platform=linux/i386 quay.io/pypa/manylinux_2_28_i686 /bin/bash -c "cd /scipy && \
-          uname -a && \
-          python3.12 -m venv test && \
-          source test/bin/activate && \
-          python -m pip install --upgrade pip && \
-          python -m pip install --group build --group dev-cli --group test-core && \
-          # we want `mpmath` but cannot install `gmpy2` on 32-bit,
-          # so cannot use `--group test-mp`
-          python -m pip install mpmath && \
-          python -m pip install -r requirements/openblas.txt && \
-          python -m pip install numpy==2.0.0 -Cbuild-dir=numpy-builddir && \
-          python -c 'import numpy as np; np.show_config()' && \
-          spin build --with-scipy-openblas=32 && \
-          spin test"
+          docker run -v "$(pwd):/scipy" --platform=linux/i386 quay.io/pypa/manylinux_2_28_i686 /bin/bash -c '
+            set -exuo pipefail
+            cd /scipy
+            uname -a
+            python3.12 -m venv test
+            source test/bin/activate
+            python -m pip install --upgrade pip
+            python -m pip install --group build --group dev-cli --group test-core
+            # we want `mpmath` but cannot install `gmpy2` on 32-bit,
+            # so cannot use `--group test-mp`
+            python -m pip install mpmath
+            python -m pip install -r requirements/openblas.txt
+            python -m pip install numpy==2.0.0 -Cbuild-dir=numpy-builddir
+            python -c "import numpy as np; np.show_config()"
+            spin build --with-scipy-openblas=32
+            spin test
+          '
 
   #################################################################################
   distro_multiple_pythons:

--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -120,6 +120,7 @@ jobs:
       run: |
         brew install llvm@19
         LLVM_PREFIX=$(brew --prefix llvm@19)
+        # shellcheck disable=SC2129
         echo CC="$LLVM_PREFIX/bin/clang" >> $GITHUB_ENV
         echo CXX="$LLVM_PREFIX/bin/clang++" >> $GITHUB_ENV
         echo LDFLAGS="-L$LLVM_PREFIX/lib" >> $GITHUB_ENV
@@ -147,8 +148,10 @@ jobs:
         # TODO
         # - add --asan to spin build command
         # - removmove -fsanitize=address from C{XX}FLAGS
-        export CFLAGS="-fsanitize=address -fno-omit-frame-pointer -fsanitize-ignorelist=$(pwd)/tools/asan-ignore.txt -Wno-deprecated-declarations"
-        export CXXFLAGS="-fsanitize=address -fno-omit-frame-pointer -fsanitize-ignorelist=$(pwd)/tools/asan-ignore.txt -Wno-deprecated-declarations"
+        CFLAGS="-fsanitize=address -fno-omit-frame-pointer -fsanitize-ignorelist=$(pwd)/tools/asan-ignore.txt -Wno-deprecated-declarations"
+        export CFLAGS
+        CXXFLAGS="-fsanitize=address -fno-omit-frame-pointer -fsanitize-ignorelist=$(pwd)/tools/asan-ignore.txt -Wno-deprecated-declarations"
+        export CXXFLAGS
         spin build -S-Dblas=accelerate
 
     - name: Test

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -76,6 +76,7 @@ jobs:
 
       - name: win_amd64 - configure rtools
         if: ${{ matrix.buildplat[1] == 'win' && matrix.buildplat[2] == 'AMD64' }}
+        shell: pwsh
         run: |
           # mingw-w64
           echo "c:\rtools45\ucrt64\bin;" >> $env:GITHUB_PATH
@@ -84,6 +85,7 @@ jobs:
         if: matrix.buildplat[1] == 'win' && matrix.buildplat[2] == 'ARM64'
         env:
           BUILDPLAT: ${{ matrix.buildplat[2] }}
+        shell: pwsh
         run: |
             echo "C:\Program Files\LLVM\bin" >> $env:GIHUB_PATH
             echo "CC=clang-cl" >> $env:GITHUB_ENV
@@ -94,6 +96,7 @@ jobs:
         if: ${{ matrix.buildplat[1] == 'win' }}
         env:
           WORKSPACE_DIR: ${{ github.workspace }}
+        shell: pwsh
         run: |
           $CIBW = "$env:WORKSPACE_DIR/.openblas"
           # pkgconfig needs a complete path, and not just "./openblas since the
@@ -117,7 +120,7 @@ jobs:
             CIBW_ENV="CIBW_ENVIRONMENT_MACOS=MACOSX_DEPLOYMENT_TARGET=14.0 INSTALL_OPENBLAS=false"
           else
             CIBW_ENV="CIBW_ENVIRONMENT_MACOS=PKG_CONFIG_PATH=$PWD/.openblas \
-              MACOSX_DEPLOYMENT_TARGET="${MACOSX_DEPLOYMENT_TARGET}""
+              MACOSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
           fi
           echo "$CIBW_ENV" >> "$GITHUB_ENV"
 

